client/asset/btc/btc.go

```diff
@@ -429,78 +572,100 @@ type findRedemptionResult struct {
 // Check that ExchangeWallet satisfies the Wallet interface.
 var _ asset.Wallet = (*ExchangeWallet)(nil)
 
+func parseChainParams(net dex.Network) (*chaincfg.Params, error) {
+	switch net {
+	case dex.Mainnet:
+		return &chaincfg.MainNetParams, nil
+	case dex.Testnet:
+		return &chaincfg.TestNet3Params, nil
+	case dex.Regtest:
+		return &chaincfg.RegressionNetParams, nil
+	}
+	return nil, fmt.Errorf("unknown network ID %v", net)
+}
+
 // NewWallet is the exported constructor by which the DEX will import the
 // exchange wallet. The wallet will shut down when the provided context is
 // canceled. The configPath can be an empty string, in which case the standard
 // system location of the bitcoind config file is assumed.
-func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network) (asset.Wallet, error) {
-	var params *chaincfg.Params
-	switch network {
-	case dex.Mainnet:
-		params = &chaincfg.MainNetParams
-	case dex.Testnet:
-		params = &chaincfg.TestNet3Params
-	case dex.Regtest:
-		params = &chaincfg.RegressionNetParams
-	default:
-		return nil, fmt.Errorf("unknown network ID %v", network)
+func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, net dex.Network) (asset.Wallet, error) {
+	params, err := parseChainParams(net)
+	if err != nil {
+		return nil, err
 	}
+
 	cloneCFG := &BTCCloneCFG{
 		WalletCFG:           cfg,
 		MinNetworkVersion:   minNetworkVersion,
 		WalletInfo:          WalletInfo,
 		Symbol:              "btc",
 		Logger:              logger,
-		Network:             network,
+		Network:             net,
 		ChainParams:         params,
 		Ports:               dexbtc.RPCPorts,
 		DefaultFallbackFee:  defaultFee,
 		DefaultFeeRateLimit: defaultFeeRateLimit,
 		Segwit:              true,
 	}
 
-	return BTCCloneWallet(cloneCFG)
+	switch cfg.Type {
+	case WalletTypeSPV:
+		if atomic.LoadUint32(&initialized) == 0 {
+			return nil, fmt.Errorf("must call asset.Initialize if you want to create an SPV wallet")
```

Comment:
Haven't encountered any doc highlighting this requirement. Since this is called from `asset.OpenWallet`, that would be a good place to indicate this requirement. Also, as part of this function's doc and the Initializer interface methods and implementations.

Is it possible to call `Initialize` here if it hasn't been called before?

Re the error message, this isn't an attempt to create an SPV wallet, more like load to me or as per the call stack leading here, open.

Reply:
Having reviewed the latest commits, some thoughts:  Please push back if you have solid reasons to do it now, @buck54321, but actually I kindly request de-scoping `Initialize` for this PR unless it's a ton more work.  Considerations (please retort if I'm seeing things wrong):

* the current function of `Initialize`, cleaning up the `logRotator` used by neutrino, does not even apply except with `WalletTypeSPV`, making it's closing more of a responsibility for a goroutine started by probably `logNeutrino` or `loadSPVWallet`/`createSPVWallet`
* the `lang` arg isn't used yet
* the purpose of the `logger` arg is unclear.  In fact it won't be used unless closing the rotator fails, so it's presence is somewhat dubious.
* it's necessary (under certain conditions and what are they?) to do asset.Initialize in the caller (Core).  In general more docs are needed to explain how the Driver is supposed to be used correctly.

ref: `(d *Driver) Initialize(ctx context.Context, wg *sync.WaitGroup, logger dex.Logger, lang language.Tag)`